ehr/src/org/labkey/ehr/table/DefaultEHRCustomizer.java

```diff
@@ -827,10 +827,10 @@ public void renderGridCellContents(RenderContext ctx, Writer out) throws IOExcep
                         Date date = (Date) ctx.get("date");
                         Object id = ctx.get(ID_COL);
 
-                        out.write("<span style=\"white-space:nowrap\"><a class=\"labkey-text-link anm-history\">[Show Hx]</a></span>");
+                        out.write("<span style=\"white-space:nowrap\"><a class=\"labkey-text-link anm-history\" data-objectid=" + objectid + " data-id=" + id + " data-date=" + date.toString() + ">[Show Hx]</a></span>");
```

Comment:
@ankurjuneja: two things:

1) Shouldnt all of the data-* attributes get quoted (line 830)? I think wrapping in jsString() would do it. 

2) This predates your PR, as far as I can tell (and I dont have every single EHR center-specific module locally), both date and objectId are always null when showClinicalHistory() is called, and they are not used for anything in that method. 

As a side point, and this also pre-dates your change, date-serialization into strings using toString() without strictly controlling format and handling of GMT offsets can give many problematic fringe cases, especially on the borders of days. That is, if one side serializes the date assuming the local time zone, but the user side parses it assuming GMT, you can get dates that shift by entire days. 

Unless there is a good reason to actually serialize them, and I dont see one based on the modules I have locally, I suggest not serializing either objectId or date here, and passing null to EHR.window.ClinicalHistoryWindow.showClinicalHistory() for both arguments. That's marginally less text written to the page and avoids date-serialization. It's also in line with every other usage I see. 

This does suggest changing the signature of showClinicalHistory() would be a good idea, but that would touch more code, and is probably out of scope for this.

Reply:
hi @bbimber , that is a good suggestion. Would be great to followup on the second point later. Can you please review again?